core/src/main/java/com/hivemq/testcontainer/core/HiveMQTestContainerCore.java

```diff
@@ -137,6 +138,20 @@ public HiveMQTestContainerCore(final @NotNull String image, final @NotNull Strin
         return self();
     }
 
+    /**
+     * Puts the supplied extension folder into '/opt/hivemq/extensions/{extension-id}' inside the container.
+     * It must at least contain a valid hivemq-extension.xml and a valid extension.jar in order to be executed.
+     * <p>
+     * Must be called before the container is started.
+     *
+     * @param extensionSupplier supplies an extension folder on the host machine
+     * @return self
+     */
+    public @NotNull SELF withExtension(final @NotNull Supplier<File> extensionSupplier) {
+        withExtension(extensionSupplier.get());
```

Comment:
Why do we add a method which takes a Supplier and then directly call get?
I would think that get would be called before each startup of HiveMQ.
For the current implementation we do not need this extra method as we already have withExtension(File). The user can call get on hisself.

Reply:
Because the withExtension method will be added fluently to a JUnit Rule/Extension and therefore the .get() will be called, when the JUnit Testclass is instantiated. So it is possible to pass code that is executed at that time (like composing the extension folder) without worrying about the JUnit Lifecylce.

Reply:
I see your point though.